sdk/version.go

```diff
@@ -0,0 +1,18 @@
+package sdk
+
+import (
+	_ "embed"
+	"strings"
+
+	"github.com/blang/semver"
+)
+
+//go:embed .version
+var version string
+
+// Version is the release version of the SDK.
```

Comment:
Should we explain here that if your using this library from a sha/head that this will be the next proposed version not something like "3.102-sha"?

Reply:
Good call, done here https://github.com/pulumi/pulumi/compare/0bfa09338e062cdb5013fc781b87d90b8746871a..d9b4d7f5afe4825953d8a0db9e9746bc3e5e9315